src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -333,6 +340,139 @@ func (r *ClusterReconciler) setLicense(
 	return nil
 }
 
+//nolint:funlen // refactor in the next iteration
+func (r *ClusterReconciler) handlePodFinalizer(
```

Comment:
do we have coverage for this? Have we also testing in cloud provider? I would be interested to know what e.g. AWS does if they decide to terminate the misbehaving node. How long it waits etc.

Reply:
As mentioned in slack, due to time constraints we do not yet have test coverage for this and have decided that manual testing is sufficient to make our deadline.

Adding automated testing is the very next thing we'll do though.